org.apache.commons.lang absence in IntelliJ 2026.2

Hi @Haehnchen! PhpStorm team is here.

The plugin starts causing `NoClassDefFoundError` exceptions in PhpStorm 2026.2 EAP releases:

```
Caused by: java.lang.NoClassDefFoundError: org/apache/commons/lang/StringUtils
	at de.espend.idea.php.phpunit.utils.PhpUnitPluginUtil.isTestClassWithoutIndexAccess(PhpUnitPluginUtil.java:80)
	at de.espend.idea.php.phpunit.type.SetUpTypeProvider.getType(SetUpTypeProvider.java:45)
	at com.jetbrains.php.lang.psi.resolve.types.PhpTypeInfo.lambda$getTypeFromAST$0(PhpTypeInfo.java:114)
```

IntelliJ 2026.2 no longer bundles `org.apache.commons.lang`, while `org.apache.commons.lang3` is still available.

Please consider migrating `org.apache.commons.lang` usages to `org.apache.commons.lang3` ones.

--- a/src/main/java/de/espend/idea/php/phpunit/annotator/MockeryAnnotator.java
+++ b/src/main/java/de/espend/idea/php/phpunit/annotator/MockeryAnnotator.java
@@ -13,7 +13,7 @@ import com.jetbrains.php.lang.psi.elements.PhpNamedElement;
 import com.jetbrains.php.lang.psi.elements.StringLiteralExpression;
 import de.espend.idea.php.phpunit.utils.MockeryReferencingUtil;
 import de.espend.idea.php.phpunit.utils.PatternUtil;
-import org.apache.commons.lang.StringUtils;
+import org.apache.commons.lang3.StringUtils;
 import org.jetbrains.annotations.NotNull;
 
 import java.util.HashSet;

--- a/src/main/java/de/espend/idea/php/phpunit/ide/PluginErrorReporterSubmitter.java
+++ b/src/main/java/de/espend/idea/php/phpunit/ide/PluginErrorReporterSubmitter.java
@@ -16,7 +16,7 @@ import com.intellij.openapi.progress.Task;
 import com.intellij.openapi.project.Project;
 import com.intellij.openapi.ui.Messages;
 import com.intellij.util.Consumer;
-import org.apache.commons.lang.StringUtils;
+import org.apache.commons.lang3.StringUtils;
 import org.apache.http.NameValuePair;
 import org.apache.http.client.methods.CloseableHttpResponse;
 import org.apache.http.client.methods.HttpPost;

--- a/src/main/java/de/espend/idea/php/phpunit/inspection/DeprecatedMockedMethodInspection.java
+++ b/src/main/java/de/espend/idea/php/phpunit/inspection/DeprecatedMockedMethodInspection.java
@@ -11,7 +11,7 @@ import com.jetbrains.php.lang.psi.elements.*;
 import de.espend.idea.php.phpunit.utils.PatternUtil;
 import de.espend.idea.php.phpunit.utils.PhpElementsUtil;
 import de.espend.idea.php.phpunit.utils.PhpUnitPluginUtil;
-import org.apache.commons.lang.StringUtils;
+import org.apache.commons.lang3.StringUtils;
 import org.jetbrains.annotations.NotNull;
 
 import java.util.Objects;

--- a/src/main/java/de/espend/idea/php/phpunit/intention/ConstructorMockIntention.java
+++ b/src/main/java/de/espend/idea/php/phpunit/intention/ConstructorMockIntention.java
@@ -20,7 +20,7 @@ import com.jetbrains.php.lang.psi.elements.*;
 import com.jetbrains.php.lang.psi.resolve.types.PhpType;
 import de.espend.idea.php.phpunit.PhpUnitIcons;
 import de.espend.idea.php.phpunit.utils.PhpElementsUtil;
-import org.apache.commons.lang.StringUtils;
+import org.apache.commons.lang3.StringUtils;
 import org.jetbrains.annotations.Nls;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;

--- a/src/main/java/de/espend/idea/php/phpunit/intention/MethodExceptionIntentionAction.java
+++ b/src/main/java/de/espend/idea/php/phpunit/intention/MethodExceptionIntentionAction.java
@@ -21,7 +21,7 @@ import com.jetbrains.php.phpunit.PhpUnitUtil;
 import de.espend.idea.php.phpunit.PhpUnitIcons;
 import de.espend.idea.php.phpunit.utils.PhpUnitPluginUtil;
 import one.util.streamex.StreamEx;
-import org.apache.commons.lang.StringUtils;
+import org.apache.commons.lang3.StringUtils;
 import org.jetbrains.annotations.Nls;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;

--- a/src/main/java/de/espend/idea/php/phpunit/linemarker/RelatedTestCaseLineMarkerProvider.java
+++ b/src/main/java/de/espend/idea/php/phpunit/linemarker/RelatedTestCaseLineMarkerProvider.java
@@ -10,7 +10,7 @@ import com.jetbrains.php.PhpIndex;
 import com.jetbrains.php.lang.psi.elements.PhpClass;
 import com.jetbrains.php.phpunit.PhpUnitUtil;
 import de.espend.idea.php.phpunit.utils.PhpElementsUtil;
-import org.apache.commons.lang.StringUtils;
+import org.apache.commons.lang3.StringUtils;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 

--- a/src/main/java/de/espend/idea/php/phpunit/reference/MockeryReferenceContributor.java
+++ b/src/main/java/de/espend/idea/php/phpunit/reference/MockeryReferenceContributor.java
@@ -7,7 +7,7 @@ import com.jetbrains.php.lang.psi.elements.StringLiteralExpression;
 import de.espend.idea.php.phpunit.utils.MockeryReferencingUtil;
 import de.espend.idea.php.phpunit.utils.PatternUtil;
 import kotlin.jvm.functions.Function3;
-import org.apache.commons.lang.StringUtils;
+import org.apache.commons.lang3.StringUtils;
 import org.jetbrains.annotations.NotNull;
 
 import java.util.ArrayList;

--- a/src/main/java/de/espend/idea/php/phpunit/reference/PhpUnitCreatePartialMock.java
+++ b/src/main/java/de/espend/idea/php/phpunit/reference/PhpUnitCreatePartialMock.java
@@ -9,7 +9,7 @@ import com.jetbrains.php.lang.psi.elements.*;
 import de.espend.idea.php.phpunit.utils.PatternUtil;
 import de.espend.idea.php.phpunit.utils.PhpElementsUtil;
 import de.espend.idea.php.phpunit.utils.PhpUnitPluginUtil;
-import org.apache.commons.lang.StringUtils;
+import org.apache.commons.lang3.StringUtils;
 import org.jetbrains.annotations.NotNull;
 
 /**

--- a/src/main/java/de/espend/idea/php/phpunit/reference/PhpUnitReferenceContributor.java
+++ b/src/main/java/de/espend/idea/php/phpunit/reference/PhpUnitReferenceContributor.java
@@ -5,7 +5,7 @@ import com.intellij.util.ProcessingContext;
 import com.jetbrains.php.lang.psi.elements.StringLiteralExpression;
 import de.espend.idea.php.phpunit.utils.PatternUtil;
 import de.espend.idea.php.phpunit.utils.PhpUnitPluginUtil;
-import org.apache.commons.lang.StringUtils;
+import org.apache.commons.lang3.StringUtils;
 import org.jetbrains.annotations.NotNull;
 
 /**

--- a/src/main/java/de/espend/idea/php/phpunit/type/SetUpTypeProvider.java
+++ b/src/main/java/de/espend/idea/php/phpunit/type/SetUpTypeProvider.java
@@ -9,7 +9,7 @@ import com.jetbrains.php.lang.psi.elements.*;
 import com.jetbrains.php.lang.psi.resolve.types.PhpType;
 import com.jetbrains.php.lang.psi.resolve.types.PhpTypeProvider4;
 import de.espend.idea.php.phpunit.utils.PhpUnitPluginUtil;
-import org.apache.commons.lang.StringUtils;
+import org.apache.commons.lang3.StringUtils;
 import org.jetbrains.annotations.Nullable;
 
 import java.util.ArrayList;

--- a/src/main/java/de/espend/idea/php/phpunit/type/utils/PhpTypeProviderUtil.java
+++ b/src/main/java/de/espend/idea/php/phpunit/type/utils/PhpTypeProviderUtil.java
@@ -5,7 +5,7 @@ import com.intellij.psi.PsiElement;
 import com.jetbrains.php.PhpIndex;
 import com.jetbrains.php.lang.psi.elements.*;
 import de.espend.idea.php.phpunit.utils.PhpElementsUtil;
-import org.apache.commons.lang.StringUtils;
+import org.apache.commons.lang3.StringUtils;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 

--- a/src/main/java/de/espend/idea/php/phpunit/utils/PhpElementsUtil.java
+++ b/src/main/java/de/espend/idea/php/phpunit/utils/PhpElementsUtil.java
@@ -13,7 +13,7 @@ import com.jetbrains.php.lang.lexer.PhpTokenTypes;
 import com.jetbrains.php.lang.psi.elements.*;
 import com.jetbrains.php.lang.psi.resolve.types.PhpType;
 import com.jetbrains.php.refactoring.PhpAliasImporter;
-import org.apache.commons.lang.StringUtils;
+import org.apache.commons.lang3.StringUtils;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 

--- a/src/main/java/de/espend/idea/php/phpunit/utils/PhpUnitPluginUtil.java
+++ b/src/main/java/de/espend/idea/php/phpunit/utils/PhpUnitPluginUtil.java
@@ -27,7 +27,7 @@ import com.jetbrains.php.lang.psi.PhpPsiElementFactory;
 import com.jetbrains.php.lang.psi.elements.*;
 import com.jetbrains.php.phpunit.PhpUnitRuntimeConfigurationProducer;
 import de.espend.idea.php.phpunit.utils.processor.CreateMockMethodReferenceProcessor;
-import org.apache.commons.lang.StringUtils;
+import org.apache.commons.lang3.StringUtils;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 

